organizing_hub/views/call_views.py

```diff
@@ -121,6 +129,46 @@ def can_make_call_for_campaign(user, call_campaign):
     return False
 
 
+def get_or_create_callers(caller_emails):
+    """
+    Get or create callers from a list of caller emails
+
+    Parameters
+    ----------
+    emails : list of str
+        List of email addresses
+
+    Returns
+        -------
+        Callers list
+            Returns list of ints corresponding to callprofile IDs
+    """
+
+    caller_ids = []
+
+    if caller_emails is not None and caller_emails != '':
+        caller_emails = [email.strip() for email in caller_emails.split(",")]
+
+        for email in caller_emails:
+            # TODO: support multiple accounts per email
+            (user, created) = User.objects.get_or_create(
+                email=email,
+                defaults={
+                    'username':email
+                }
+            )
+
+            if not hasattr(user,'bsdprofile'):
+                BSDProfile.objects.create(user=user)
+
+            if not hasattr(user,'callprofile'):
+                CallProfile.objects.create(user=user)
+
+            caller_ids.append(user.callprofile.id)
```

Comment:
I get confused about this a lot, but will this return stale data sometimes? You already have the `user` object from earlier, and you just created a call profile now, so will `user.callprofile` return a call profile that was just created? might want to just actually pull the newly created call profile from the create method and return call profile id that way?

Reply:
I don't follow. We're creating or getting a user, and then if they don't have a call profile we're creating one, and adding it to list. If they already have a call profile they'll be added to list. Are you saying if a call profile is created elsewhere while this is happening? A user should only ever have one call profile. I don't follow `might want to just actually pull the newly created call profile from the create method and return call profile id that way?`

Reply:
I'm saying that when you initially get the `user` it might not have a call profile attached. So you create a call profile. But you do not clearly refresh the `user` so it might still say there is no call profile until you recheck the db?

Reply:
So you mean do something like this?
```if not hasattr(user,'callprofile'):
                callprofile = CallProfile.objects.create(user=user)

            caller_ids.append(callprofile.id)```

Reply:
yeah basically. if it's true that user object will not auto updated without a data refresh, then need something like ```
            if hasattr(user, 'callprofile'):
                call_profile = user.callprofile
            else:
                call_profile = CallProfile.objects.create(user=user)
             caller_ids.append(callprofile.id)
```

Reply:
In that case, I think get_or_update would work here?

Reply:
I think you mean `get_or_create`? Yeah you could try that, but then you cant use the `user.callprofile` syntax which is convenient. But yeah essentially the code snippet above is very similar to `get_or_create`.